llm-chain/src/tools/collection.rs

```diff
@@ -21,28 +20,33 @@ pub enum ToolUseError {
     InvalidInvocation,
     #[error("Tool invocation failed: {0}")]
     ToolInvocationFailed(String),
+    #[error(transparent)]
+    ToolError(#[from] E),
 }
 
-impl ToolCollection {
+impl<T> ToolCollection<T>
```

Comment:
What's the purpose of the collection with this?

Reply:
Good question, if it were possible to construct a Tool using only it's name, just like we are able to with BashTool, ExitTool an PythonTool, then we could remove the ToolCollection entirely and construct the Multitool on the fly. But I'm assuming we would like to store some dependencies for each Tool, so it seems to me like we will need some structure that will hold Tools that are already initialized with all their dependencies.

Reply:
Okay yeah that makes sense to me now :)